docs/source/user_guide/key_concepts.md

```diff
@@ -0,0 +1,202 @@
+See the [API documentation](api_index) for full reference.
+
+## How derotation works
+
+Without derotation, movies acquired with a line scanning microscope are geometrically distorted — each line is captured at a slightly different angle, making it hard to register or interpret the resulting images.
+
+If the angle of rotation is recorded, `derotation` can reconstruct each frame by assigning a rotation angle to each acquired line and rotating it back to its original position. This is what is called derotation-by-line. This process incrementally reconstructs each frame and can optionally include shear deformation correction.
+
+![Derotation process](../_static/derotation_by_line.gif)
+
+## How to use derotation
+
+There are two main ways to use `derotation`:
+
+### 1. **Low-level core function**
+Use `derotate_an_image_array_line_by_line` to derotate an image stack, given:
+- The original calcium imaging movie (expects only one imaging plane)
+- Rotation angle per line
+
+This is ideal for testing and debugging with synthetic or preprocessed data.
+
+### 2. **Full or incremental pipeline classes**
+Use the pre-made pipelines to run end-to-end processing:
+
+- `FullPipeline`: assumes randomised complete clockwise and counter-clockwise rotations. It includes:
+  - Analog signal parsing
+  - Angle interpolation
+  - Bayesian optimization for center estimation
+  - Derotation by line (calling `derotate_an_image_array_line_by_line`)
+
+- `IncrementalPipeline`: assumes a continuous rotation performed in small increments. It inherits functionality from the `FullPipeline` but does not perform Bayesian optimization. It can be useful as an alternative way to estimate center of rotation.
+
+Both pipelines accept a configuration dictionary (see the [configuration guide](configuration)) and output:
+- Derotated TIFF
+- CSV with rotation angles and metadata
+- Debugging plots
+- A text file with optimal center of rotation
+- Logs
+
+You can subclass `FullPipeline` to create custom pipelines by overwriting relevant methods.
+
+### Example usage of FullPipeline:
+
+```python
+from derotation.config.load_config import update_config_paths, load_config
+from derotation.analysis.full_derotation_pipeline import FullPipeline
+
+
+# Load the configuration file
+pipeline_type = "full"
+config = load_config(pipeline_type)
+config = update_config_paths(
+    config=config,
+    tif_path="/my/data/image_stack.tif",
+    bin_path="/my/data/signals.bin",
+    dataset_path="/my/data/",
+    kind=pipeline_type,
+)
+
+# Load data and run the pipeline
+pipeline = FullPipeline(config)
+pipeline()
+```
+
+---
+
+## Finding the center of rotation
+
+An accurate center of rotation is crucial for high-quality derotation. Even small errors can produce residual motion in the derotated movie — often visible as circles traced by stationary objects as cells.
+
+![Residual motion](../_static/wrong_center.png)
```

Comment:
Sphinx has some nice functionality for image captions which would look nice here.

Reply:
Have to look at them thaks!